ruma-events/src/lib.rs

```diff
@@ -237,6 +241,47 @@ pub trait MessageEventContent: RoomEventContent {}
 /// Marker trait for the content of a state event.
 pub trait StateEventContent: RoomEventContent {}
 
+/// The base trait that all redacted event content types implement.
+///
+/// Implementing this trait allows content types to be serialized as well as deserialized.
+pub trait RedactedEventContent: EventContent {
+    /// Determines if the redacted event content needs to serialize fields.
+    fn has_serialize_fields(&self) -> bool;
+
+    /// Determines if the redacted event content needs to deserialize fields.
+    fn has_deserialize_fields() -> bool;
+
+    /// Constructs the redacted event content.
+    ///
+    /// If called for anything but "empty" redacted content this will error.
+    fn redacted(_event_type: &str) -> Result<Self, serde_json::Error> {
```

Comment:
I wanted to suggest renaming this to `empty` and having it as the first function in the trait two times now. Somehow I failed both times, due to not finding the right place to put the annotation in the diff and then forgetting. Oh well, third time lucky.